### ISSUE TITLE
[Bug]: PaddedBox not working

### ISSUE THREAD

**Issue by oscargus** [matplotlib/matplotlib#24620]
### Bug summary
 
`PaddedBox` https://matplotlib.org/stable/api/offsetbox_api.html#matplotlib.offsetbox.PaddedBox doesn't look like it is working properly.

### Code for reproduction

```python
import matplotlib.pyplot as plt
from matplotlib.offsetbox import PaddedBox, AnchoredText

fig, ax = plt.subplots()

at = AnchoredText("foo",  'upper left')
pb = PaddedBox(at, pad=1, patch_attrs={'facecolor' : 'r'}, draw_frame=True)

ax.add_artist(pb)
fig.show()
```

### Actual outcome

![paddedbox](https://user-images.githubusercontent.com/8114497/205509015-74482d22-b23e-44ab-8a7f-375c138bc865.png)

### Expected outcome

Red box is at same location as text.

### Additional information

It seems like the sizing is working OK (using a longer string will change the size accordingly), but that the positioning is not working.

Also, not passing `pad` leads to an error, so it should probably be 0. as default rather than `None`.

### Matplotlib Version

`main`

### Installation

git checkout

**Comment by oscargus** [comment]
I guess that it may be good with some sort of clarification how the hierarchy of offset boxes works, so adding the doc label.

If I understand it correctly, some, like `TextArea` and `DrawingArea` are the "base artists". These can be put in `PaddedBox` and `Packers`. Any of the earlier can then be put in `Anchored*`?

**Comment by leejjoon** [comment]
Correct. `OffsetBox` knows it size but not its position. Its parent should set its position at the drawing time. So, the top-most parent of offsetboxes should be `Anchored*` or `AnnotationBbox` that know how to position their children.
